File: app/src/main/java/com/tarashor/currencyconverter/BaseApplication.java
package com.tarashor.currencyconverter;

import android.app.Application;
import com.tarashor.currencyconverter.di.DaggerAppComponent;
import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

//public class BaseApplication extends DaggerApplication {
//
//    @Override
//    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
//        return DaggerAppComponent.builder().application(this).build();
//    }
//}

public class BaseApplication extends Application {


}

File: app/src/main/java/com/tarashor/currencyconverter/BaseApplication.kt
package com.tarashor.currencyconverter

import androidx.appcompat.app.AppCompatActivity
import android.app.Application
import com.tarashor.currencyconverter.di.AppComponent
import com.tarashor.currencyconverter.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector
import javax.inject.Inject

class BaseApplication : Application(), HasAndroidInjector {

    private lateinit var appComponent: AppComponent

    @Inject
    lateinit var androidInjector: DispatchingAndroidInjector<Any>


    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerAppComponent.builder().application(this).build()
        appComponent.inject(this)
    }


    override fun androidInjector(): AndroidInjector<Any> = androidInjector
}
